frontend/src/components/StockData/StockData.jsx

```diff
@@ -1,38 +1,74 @@
-import './StockData.css'
-import { useState, useEffect } from 'react'
-import axios from 'axios'
+import "./StockData.css";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { ALPHA_API_KEY, FINNHUB_API_KEY } from "../../constants";
 
 export default function StockData({ currentStock }) {
-    const API_KEY_1 = "5DIHMIONQPZFEG6R";
-    const API_KEY_2 = "ciefl6pr01qmfas4d2hgciefl6pr01qmfas4d2i0";
     const [stockData, setStockData] = useState({});
-    const [stockPrice, setStockPrice] = useState({});
+    const [isLoading, setIsLoading] = useState(true);
+    const [stockNotFound, setStockNotFound] = useState(false);
 
-    useEffect( () => {
+    useEffect(() => {
         const fetchStockData = async () => {
             try {
-                const infoUrl = `https://www.alphavantage.co/query?function=OVERVIEW&symbol=${currentStock}&apikey=${API_KEY_1}`;
-                const priceUrl = `https://finnhub.io/api/v1/quote?symbol=${currentStock}&token=${API_KEY_2}`;
-                const responseInfo = await axios.get(infoUrl);
-                const responsePrice = await axios.get(priceUrl)
-                const infoData = responseInfo.data;
-                const priceData = responsePrice.data;
-                setStockData(infoData);
-                setStockPrice(priceData);
-            } catch(error) {
-                
+                setIsLoading(true);
+
+                const databaseResponse = await axios.get(`http://localhost:3000/stocks/${currentStock}`);
+                const stockExistsInDatabase = databaseResponse.data ? true : false;
+
+                if (stockExistsInDatabase) {
+                    setStockData(databaseResponse.data);
+                } else {
```

Comment:
To avoid deeper nesting, you can use early return:
```
if (stockExistsInDatabase) {
  setStockData(databaseResponse.data);
  return;
}

const stockOverviewUrl = `https://www.alphavantage.co/query?
// the rest of the code in the function
```